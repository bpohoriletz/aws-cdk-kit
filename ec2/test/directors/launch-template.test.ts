import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import LaunchTemplateDirector from '../../directors/launch-template';
import Al2023PublicBuilder from '../../launch-template-builders/al2023-public-builder';

describe('new LaunchTemplateDirector()', () => {
  let instance: LaunchTemplateDirector;
  let stack: cdk.Stack;

  beforeEach(() => {
    instance = new LaunchTemplateDirector(Al2023PublicBuilder);
    stack = new cdk.Stack();
  });

  describe('#constructEc2Group', () => {
    beforeEach(() => {});

    it('uses defaults if only name is passed', () => {
      instance.constructPublicEc2LaunchTemplate(stack, 'Ec2LaunchTemplate');

      expect(Template.fromStack(stack).toJSON().Resources).toMatchSnapshot();
    });
  });
});
