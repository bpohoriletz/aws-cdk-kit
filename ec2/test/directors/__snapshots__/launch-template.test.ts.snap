// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new LaunchTemplateDirector() #constructEc2Group uses defaults if only name is passed 1`] = `
{
  "Ec2LaunchTemplate37DE02F5": {
    "Properties": {
      "LaunchTemplateData": {
        "ImageId": {
          "Ref": "SsmParameterValueawsserviceamiamazonlinuxlatestal2023amikernel61x8664C96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "NetworkInterfaces": [
          {
            "AssociatePublicIpAddress": true,
            "DeviceIndex": 0,
          },
        ],
        "TagSpecifications": [
          {
            "ResourceType": "instance",
            "Tags": [
              {
                "Key": "Name",
                "Value": "Default/Ec2LaunchTemplate",
              },
            ],
          },
          {
            "ResourceType": "volume",
            "Tags": [
              {
                "Key": "Name",
                "Value": "Default/Ec2LaunchTemplate",
              },
            ],
          },
        ],
        "UserData": {
          "Fn::Base64": "#!/bin/bash
echo Hello World",
        },
      },
      "TagSpecifications": [
        {
          "ResourceType": "launch-template",
          "Tags": [
            {
              "Key": "Name",
              "Value": "Default/Ec2LaunchTemplate",
            },
          ],
        },
      ],
    },
    "Type": "AWS::EC2::LaunchTemplate",
  },
}
`;
